tests/page/matchers.misc.spec.ts

```diff
@@ -23,3 +24,23 @@ it('should outlive frame navigation', async ({ page, server }) => {
   }, 1000);
   await expect(page.locator('.box').first()).toBeEmpty();
 });
+
+it('should print no-locator-resolved error when locator matcher did not resolve to any element', async ({ page, server }) => {
+  const myLocator = page.locator('.nonexisting');
+  const expectWithShortLivingTimeout = expect.configure({ timeout: 10 });
+  const locatorMatchers = [
+    () => expectWithShortLivingTimeout(myLocator).toBeAttached(),                 // Boolean matcher
+    () => expectWithShortLivingTimeout(myLocator).toHaveJSProperty('abc', 'abc'), // Equal matcher
+    () => expectWithShortLivingTimeout(myLocator).not.toHaveText('abc'),          // Text matcher - pass / string
+    () => expectWithShortLivingTimeout(myLocator).not.toHaveText(/abc/),          // Text matcher - pass / RegExp
+    () => expectWithShortLivingTimeout(myLocator).toContainText('abc'),           // Text matcher - fail
```

Comment:
Perhaps test both fail/string and fail/RegExp?

Reply:
(Same branch, but sure added it)